pkg/upload/js.go

```diff
@@ -16,9 +24,163 @@ type JsOptions struct {
 	Path        utils.Paths `arg:"" name:"path" help:"Path to a directory of source maps and bundles to upload" type:"path" default:"."`
 }
 
+// Resolve the project if it isn't specified using the currend working directory
+func resolveProjectRoot(projectRoot string, path string) string {
+	if projectRoot != "" {
+		return projectRoot
+	}
+	workingDirectory, err := os.Getwd()
+	if err != nil {
+		return path
+	}
+	return workingDirectory
+}
+
+// Attempt to parse information from the package.json file if values aren't provided on the command line
+func ResolveVersion(versionName string, projectRoot string, logger log.Logger) (string, error) {
+	if versionName != "" {
+		return versionName, nil
+	}
+	checkPath, err := filepath.Abs(projectRoot)
+	if err != nil {
+		return "", fmt.Errorf("unable to make project root an absolute path %s", projectRoot)
+	}
+	// Walk up the folder structure as far as possible
+	for filepath.Dir(checkPath) != checkPath {
+		packageJson := filepath.Join(checkPath, "package.json")
```

Comment:
Does this work without erroring if we don't find a package.json and/or if you don't have permission to ascend above a certain level?

It would be worth adding a test where we don't have a package.json to ensure it doesn't error out.

Reply:
The filepath.join is a string operation and will never fail. If the file is not found or the permissions are not correct then the file.open will fail. This error is caught.

In Go all of the recoverable errors must be passed up as return values and handled.

Reply:
Adding a test for this is difficult because there is a package.json in the root of this project and so if the fixture didn't have one then the root one would be used.